libs/lib_parsing/Origin.mli

```diff
@@ -59,13 +63,26 @@
       {- [Stdin] origins are for content read from standard input.}
       {- [Network uri] origins are for content obtained from [uri] over the
       network.}
-      {- [GitBlob info] origins are for git blob objects with metadata given in
-      [info] (e.g., the blob's sha, commits the blob is present at, etc..).}
     }
  *)
-type t = File of Fpath.t [@@deriving show, eq, ord]
-(* TODO: Evaluate futher using Ppath.t instead of Fpath.t, since it documents
-   we want this to be a relative path from the project root. *)
+type t =
+  | File of Fpath.t
+  (* TODO: Evaluate futher using Ppath.t instead of Fpath.t, since it documents
+     we want this to be a relative path from the project root. *)
+  | GitBlob of {
+      sha : Git_wrapper.sha;
+          (** The hash for this specific git {e blob}. Used for git operations and to
+              identify the object. *)
+      paths : (Git_wrapper.sha * Fpath.t) list;
```

Comment:
> how many paths do we expect for a given blob? if i have a repo with thousands of commits, will this be a list thousands of entries long?

Yes, this could potentially be rather large.

> i notice that we only use it in the case where we filter for includes and excludes. just wondering if it might be helpful to make this lazy, if that code path is not exercised

I don't think so: (1) with the way this information is gathered we would basically be making one lazy value which would shared between all git object targets (since we batch collect this information); (2) if at least one rule uses this, we suddenly force it for everything, making benchmarking harder, plus we have to carry around the lazy value until then. I think it more realistic to look into strategies to minimize duplication iff it becomes an issue (but I don't see value in potentially premature optimization here).